app/Http/Controllers/ChatController.php

```diff
@@ -26,19 +26,17 @@ public function index()
 
         $json = [
             'last_message_id' => optional(Message::last())->getKey(),
-            'presence' => UserChannel::presenceForUser($user),
+            'presence' => (new UserChannelList($user))->get(),
         ];
 
         $targetUser = User::lookup(Request::input('sendto'), 'id');
         if ($targetUser) {
-            $canMessage = priv_check('ChatStart', $targetUser)->can();
-
             $channel = Channel::findPM($targetUser, $user);
-            optional($channel)->addUser($user);
+            $channel?->addUser($user);
 
             $json['send_to'] = [
-                'can_message' => $canMessage,
-                'channel_id' => optional($channel)->getKey(),
+                'can_message' => $channel?->canMessage($user) ?? false,
```

Comment:
haven't checked but this seem to be always false for new channels? (`?sendto=user` where there's no pm channel with the `user` yet)